aggregate/billing_functions/utils.py

```diff
@@ -0,0 +1,907 @@
+# pylint: disable=global-statement,too-many-arguments,line-too-long
+"""
+Class of helper functions for billing aggregate functions
+"""
+import os
+import re
+import math
+import json
+import logging
+from pathlib import Path
+from io import StringIO
+from collections import defaultdict
+from datetime import datetime, timedelta
+import sys
+from typing import Any, Iterator, Sequence, TypeVar, Iterable
+
+import asyncio
+import aiohttp
+import pandas as pd
+import google.cloud.bigquery as bq
+
+from google.cloud import secretmanager
+from google.api_core.exceptions import ClientError
+
+
+logger = logging.getLogger('Cost Aggregate')
+
+handler = logging.StreamHandler(sys.stderr)
+handler.setFormatter(logging.Formatter('%(asctime)s %(levelname)s: %(message)s'))
+
+logger.addHandler(handler)
+if os.getenv('DEV') in ('1', 'true', 'yes'):
+    logger.setLevel(logging.DEBUG)
+
+# pylint: disable=invalid-name
+T = TypeVar('T')
+
+DEFAULT_TOPIC = 'admin'
+
+GCP_PROJECT = 'billing-admin-290403'
+
+GCP_BILLING_BQ_TABLE = (
+    f'{GCP_PROJECT}.billing.gcp_billing_export_v1_01D012_20A6A2_CBD343'
+)
+GCP_AGGREGATE_DEST_TABLE = os.getenv(
+    'GCP_AGGREGATE_DEST_TABLE', 'billing-admin-290403.billing_aggregate.aggregate'
+)
+
+assert GCP_AGGREGATE_DEST_TABLE
+logger.info(f'GCP_AGGREGATE_DEST_TABLE: {GCP_AGGREGATE_DEST_TABLE}')
+
+IS_PRODUCTION = os.getenv('PRODUCTION') in ('1', 'true', 'yes')
+
+# 10% overhead
+HAIL_SERVICE_FEE = 0.05
+DEFAULT_BQ_CHUNK_SIZE = 9000
```

Comment:
Is this the page size used for results?

Reply:
This is used to chunk inserts into BQ, as you can only have 10,000 parameters in a standard sql param: https://cloud.google.com/bigquery/quotas#:~:text=A%20Standard%20SQL%20query%20can%20have%20up%20to%2010%2C000%20parameters - I've renamed and added a comment.